c_src/lin.cpp

```diff
@@ -754,3 +918,146 @@ void erl2lin_free_model (LINEAR_MODEL* model)
 void lin_print (const char* message) {
    // suppress debug output
 }
+/*-----------< FUNCTION: lin_calibrate_train >-------------------------------
+// Purpose:    calibrates decision outputs to class probabilities using
+//             univariate binary logistic regression
+//             lifted from libsvm sigmoid_train, which is not exported
+// Parameters: m        - number of training examples
+//             decision - decision ouputs for each training example
+//                        used as X in the calibration model
+//             labels   - class labels for each training example (pos/neg)
+//             prob_a   - return regression slope parameter via here
+//             prob_b   - return regression intercept parameter via here
+// Returns:    none
+---------------------------------------------------------------------------*/
+void lin_calibrate_train (
+   int     m,
+   double* decision,
+   double* labels,
+   double& prob_a,
+   double& prob_b)
+{
+   double prior1=0, prior0 = 0;
+   int i;
+
+   for (i=0;i<m;i++)
+      if (labels[i] > 0) prior1+=1;
+      else prior0+=1;
+
+   int max_iter=100; // Maximal number of iterations
```

Comment:
Should this and `min_step` be configurable somewhere to retrain in the event of failure?

Reply:
I'd rather keep this identical with the LIBSVM version, including the bad style.